packages/dds/tree/src/core/tree/detachedFieldIndex.ts

```diff
@@ -35,15 +38,43 @@ import { IIdCompressor } from "@fluidframework/id-compressor";
  */
 export type ForestRootId = Brand<number, "tree.ForestRootId">;
 
+/**
+ * fake revision used to mark that the revision stored in a {@link DetachedFieldIndex} is not yet
+ * set after loading data from a summary
+ */
+export const fakeRevisionWhenNotSet = 33 as SessionSpaceCompressedId;
+
 /**
  * The tree index records detached field IDs and associates them with a change atom ID.
  */
 export class DetachedFieldIndex {
-	// TODO: don't store the field key in the index, it can be derived from the root ID
-	private detachedNodeToField: NestedMap<Major, Minor, ForestRootId> = new Map();
+	/**
+	 * A mapping from detached node ids to forest root ids and the revision that last created or modified the root.
+	 *
+	 * @remarks
+	 * undefined revisions are tolerated but any roots not associated with a revision must be disposed manually
+	 */
+	private detachedNodeToField: NestedMap<
+		Major,
+		Minor,
+		{ readonly root: ForestRootId; latestRelevantRevision: RevisionTag | undefined }
```

Comment:
It seems risky not to make `latestRelevantRevision` readonly. You do use functions (like `populateNestedMap`) that leads to these entries being shared across multiple detached fields, so mutating some of them could cause very subtle bugs.